commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -118,17 +140,81 @@ void testAddAndAddAll() {
 
         //Test for adding an empty list to an empty list
         ComplexList list = ComplexList.interleaved();
-        assertListOperation(l -> {
+        assertListOperation1(l -> {
+            l.addAll(list);
+            return l.addAll(0, list);
+        });
+    }
+
+    @Test
+    void testAddAndAddAllForComplexNonInterleavedList() {
```

Comment:
Code duplication of the entire method for a new list is not maintainable. Since the list must pass the same test then you can do this using a parameterized test:
```Java
    /**
     * Generate a stream of arguments containing empty {@code Complex<List>} implementations.
     *
     * @return the stream of arguments
     */
    static Stream<Arguments> listImplementations() {
        return Stream.of(Arguments.of(ComplexList.interleaved()),
                         Arguments.of(ComplexList.nonInterleaved()));
    }

    @ParameterizedTest
    @MethodSource({"listImplementations"})
    void testAddAndAddAllList(List<Complex> l2) {
        List<Complex> l1 = new ArrayList<>();
        assertListOperation(list -> list.add(Complex.ofCartesian(1, 2)), l1, l2);
        assertListOperation(list -> {
            list.add(1, Complex.ofCartesian(10, 20));
            return Boolean.TRUE;
        }, l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(13, 14)), l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(15, 16)), l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(17, 18)), l1, l2);
        assertListOperation(list -> {
            list.addAll(1, list);
            return Boolean.TRUE;
        }, l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(19, 20)), l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(21, 22)), l1, l2);
        assertListOperation(list -> list.add(Complex.ofCartesian(23, 24)), l1, l2);
    }
```
The test may have to be broken down into smaller tests as the parameter will be an empty list. If you wish to start with a list populated with e.g. 10 random Complex numbers then you can add a second static method to stream arguments with populated lists.

Note: You can create multiple parameters for the parameterized test if these require:
```Java
    static Stream<Arguments> listImplementationsWithSize() {
        return Stream.of(Arguments.of(ComplexList.interleaved(), 10),
                         Arguments.of(ComplexList.nonInterleaved(), 10));
    }

    @ParameterizedTest
    @MethodSource({"listImplementationsWithSize"})
    void testAddAndAddAllList(List<Complex> l2, int size) {
```